dot/parachain/provisioner/provisioner_test.go

```diff
@@ -1495,3 +1495,94 @@ func hashFromU64BE(val uint64) common.Hash {
 	copy(h[common.HashLength-8:], b)
 	return h
 }
+
+func TestRequestInherentData(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name  string
+		perRP map[common.Hash]*perRelayParent
+		//msg                provisionermessages.RequestInherentData
+		mockBlockState        func(*gomock.Controller) BlockState
+		expectErr             bool
+		expectAwaitInherit    bool
+		expectAwaitInheritLen int
+	}{
+		{
+			name:  "unknown_relay_parent",
+			perRP: map[common.Hash]*perRelayParent{},
+			mockBlockState: func(ctrl *gomock.Controller) BlockState {
+				return NewMockBlockState(ctrl)
+			},
+			expectErr: false,
+		},
+		{
+			name: "inherent_not_ready",
+			perRP: map[common.Hash]*perRelayParent{
+				{1}: {
+					leaf:            &parachaintypes.ActivatedLeaf{Hash: common.Hash{1}},
+					isInherentReady: false,
+				},
+			},
+			mockBlockState: func(ctrl *gomock.Controller) BlockState {
+				return NewMockBlockState(ctrl)
+			},
+			expectErr:          false,
+			expectAwaitInherit: true,
+		},
+		{
+			// This case triggers the sendInherentData method which has its own unit test.
+			// To avoid redundant testing, we simulate a GetRuntime error here.
+			name: "inherent_ready",
+			perRP: map[common.Hash]*perRelayParent{
+				{1}: {
+					leaf:            &parachaintypes.ActivatedLeaf{Hash: common.Hash{1}},
+					isInherentReady: true,
+					signedBitfields: []parachaintypes.CheckedSignedAvailabilityBitfield{},
+				},
+			},
+			mockBlockState: func(ctrl *gomock.Controller) BlockState {
+				bs := NewMockBlockState(ctrl)
+				bs.EXPECT().
+					GetRuntime(common.Hash{1}).
+					Return(nil, fmt.Errorf("mock runtime error")).
+					Times(1)
+				return bs
+			},
+			expectErr: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
```

Comment:
Is this still necessary? I thought [this issue was fixed](https://go.dev/blog/loopvar-preview) a long time ago.